localrpc/setup_test.go

```diff
@@ -0,0 +1,579 @@
+	"github.com/MadBase/MadNet/constants"
+	"github.com/MadBase/MadNet/crypto"
+	mncrypto "github.com/MadBase/MadNet/crypto"
+	"github.com/MadBase/MadNet/dynamics"
+	"github.com/MadBase/MadNet/ipc"
+	"github.com/MadBase/MadNet/logging"
+	"github.com/MadBase/MadNet/peering"
+	"github.com/MadBase/MadNet/proto"
+	pb "github.com/MadBase/MadNet/proto"
+	"github.com/MadBase/MadNet/status"
+	"github.com/MadBase/MadNet/utils"
+	mnutils "github.com/MadBase/MadNet/utils"
+	"github.com/dgraph-io/badger/v2"
+	"github.com/spf13/viper"
+)
+
+var address string = "localhost:8884"
+var timeout time.Duration = time.Second * 10
+var account []byte
+var owner *objs.ValueStoreOwner
+var signer *crypto.Secp256k1Signer
+var txHash []byte
+var pubKey []byte
+var tx *objs.Tx
+var geth = exec.Cmd{}
+var err error
+var badgerDB *badger.DB
+var srpc *Handlers
+var lrpc *Client
+var tx1, tx2, tx3 *pb.TransactionData
+var tx1Hash, tx2Hash, tx3Hash []byte
+var consumedTx1Hash, consumedTx2Hash, consumedTx3Hash []byte
+var utxoTx1IDs, utxoTx2IDs, utxoTx3IDs [][]byte
+var tx1Value, tx2Value, tx3Value uint64
+var tx1Signature, tx2Signature, tx3Signature []byte
+
+// var stateDB *db.Database
+var ctx context.Context
+var app *application.Application
+var consGossipHandlers *gossip.Handlers
+var ipcServer *ipc.Server
+var peerManager *peering.PeerManager
+var consGossipClient *gossip.Client
+var consDlManager *dman.DMan
+var localStateHandler *Handlers
+var consDB *db.Database
+var consSync *consensus.Synchronizer
+var storage *dynamics.Storage
+
+func TestMain(m *testing.M) {
+
+	loadSettings("validator.toml")
+	signer, pubKey = getSignerData()
+	validatorNode()
+
+	srpc = &Handlers{
+		ctx:         ctx,
+		cancelCtx:   nil,
+		database:    consDB,
+		sstore:      nil,
+		AppHandler:  app,
+		GossipBus:   consGossipHandlers,
+		Storage:     storage,
+		logger:      nil,
+		ethAcct:     nil,
+		EthPubk:     nil,
+		safeHandler: func() bool { return true },
+		safecount:   1,
+	}
+	srpc.Init(srpc.database, srpc.AppHandler, srpc.GossipBus, srpc.EthPubk, srpc.safeHandler, srpc.Storage)
+
+	go srpc.Start()
+	defer srpc.Stop()
+
+	lrpc = &Client{
+		Mutex:       sync.Mutex{},
+		closeChan:   nil,
+		closeOnce:   sync.Once{},
+		Address:     config.Configuration.Transport.LocalStateListeningAddress,
+		TimeOut:     timeout,
+		conn:        nil,
+		client:      nil,
+		wg:          sync.WaitGroup{},
+		isConnected: false,
+	}
+	go lrpc.Connect(ctx)
+	defer lrpc.Close()
+
+	localStateServer := initLocalStateServer(srpc)
+	go localStateServer.Serve()
+	defer localStateServer.Close()
+
+	go ipcServer.Start()
+	defer ipcServer.Close()
+
+	go peerManager.Start()
+	defer peerManager.Close()
+
+	go consGossipClient.Start()
+	defer consGossipClient.Close()
+
+	go consDlManager.Start()
+	defer consDlManager.Close()
+
+	go localStateHandler.Start()
+	defer localStateHandler.Stop()
+
+	go consGossipHandlers.Start()
+	defer consGossipHandlers.Close()
+
+	go storage.Start()
```

Comment:
don't we need a `defer storage.Close()` here?

Reply:
Agree but actually there is not Close() method on Storage (or anything similar)